Fix schema migration version type

PNixx opened the pull request:
### Summary

Column `version` of table has a type `string` for `schema_migration` table. We need variable with a `string` type to insert to the table.

eileencodes commented:
Hey @pnixx thanks for the PR and welcome to Rails!

While I appreciate you taking the time to contribute to Rails this change isn't necessary because `quote` will ensure that it's a string when passed to the execute call. Hope to see other contributions from you in the future!

PNixx commented:
I create gem clickhouse-activerecord https://github.com/PNixx/clickhouse-activerecord/blob/master/lib/active_record/connection_adapters/clickhouse/schema_statements.rb#L54-L63. The database clickhouse causes an error when trying to insert data with the wrong data type. Method `quote` only add `"` for strings, but you send integer type.
Example request and error without convertion type:
```
ActiveRecord::StatementInvalid: ActiveRecord::ActiveRecordError: Response code: 500:
Code: 53, e.displayText() = DB::Exception: Type mismatch in IN or VALUES section. Expected: String. Got: UInt64, e.what() = DB::Exception
: INSERT INTO schema_migrations (version) VALUES (20190305142553)
```
Table structure:
```
DESCRIBE TABLE schema_migrations

┌─name────┬─type───┬─default_type─┬─default_expression─┬─comment─┬─codec_expression─┐
│ version │ String │              │                    │         │                  │
└─────────┴────────┴──────────────┴────────────────────┴─────────┴──────────────────┘
```

For me, the best solution would be not to override the method `assume_migrated_upto_version`. Therefore, I sent a pull request.

eileencodes commented:
What database are you using? Locally my schema_migrations for mysql is a varchar.

PNixx commented:
I using a ClickHouse database https://clickhouse.yandex/

eileencodes commented:
Discussed internally and we have decided this change is 👍 but we need a test added to prevent us accidentally removing this in the future. Can you add that?

PNixx committed:
Fix schema migration version type
Column `version` of table has a type `string`. We need variable with a `string` type to insert to the table.
